### ISSUE TITLE
brew install python3 error in make file

### ISSUE THREAD

**Issue by rixj** [sphinx-doc/sphinx#4902]
Subject: <AttributeError: 'DirectiveAdapter' object has no attribute 'env' when trying to "brew install python3">

### Problem
- I tried to download python3 through brew.

#### Procedure to reproduce the problem
```
$ brew install python3

```

#### Error logs / results
```
==> Installing dependencies for python: sphinx-doc
==> Installing python dependency: sphinx-doc
==> Downloading https://homebrew.bintray.com/bottles/sphinx-doc-1.7.3.high_sier
==> Downloading from https://akamai.bintray.com/34/34d0333374dc4388dbc6c6637c87
######################################################################## 100.0%
==> Pouring sphinx-doc-1.7.3.high_sierra.bottle.tar.gz
==> Caveats
This formula is keg-only, which means it was not symlinked into /usr/local,
because this formula is mainly used internally by other formulae.
Users are advised to use `pip` to install sphinx-doc.

If you need to have this software first in your PATH run:
  echo 'export PATH="/usr/local/opt/sphinx-doc/bin:$PATH"' >> ~/.bash_profile

==> Summary
🍺  /usr/local/Cellar/sphinx-doc/1.7.3: 2,642 files, 36.8MB
==> Installing python
Warning: Building python from source:
  The bottle needs the Apple Command Line Tools to be installed.
  You can install them, if desired, with:
    xcode-select --install

==> Downloading https://www.python.org/ftp/python/3.6.5/Python-3.6.5.tar.xz
Already downloaded: /Users/Danielle/Library/Caches/Homebrew/python-3.6.5.tar.xz
==> ./configure --prefix=/usr/local/Cellar/python/3.6.5 --enable-ipv6 --dataroo
==> make
==> make install PYTHONAPPSDIR=/usr/local/Cellar/python/3.6.5
==> make frameworkinstallextras PYTHONAPPSDIR=/usr/local/Cellar/python/3.6.5/sh
==> Downloading https://files.pythonhosted.org/packages/72/c2/c09362ab29338413a
Already downloaded: /Users/Danielle/Library/Caches/Homebrew/python--setuptools-39.0.1.zip
==> Downloading https://files.pythonhosted.org/packages/c4/44/e6b8056b6c8f2bfd1
Already downloaded: /Users/Danielle/Library/Caches/Homebrew/python--pip-9.0.3.tar.gz
==> Downloading https://files.pythonhosted.org/packages/fa/b4/f9886517624a4dcb8
Already downloaded: /Users/Danielle/Library/Caches/Homebrew/python--wheel-0.30.0.tar.gz
==> make html
Last 15 lines from /Users/Danielle/Library/Logs/Homebrew/python/05.make:
reading sources... [ 41%] library/email.examples
reading sources... [ 41%] library/email.generator
reading sources... [ 41%] library/email.header
reading sources... [ 42%] library/email.headerregistry
reading sources... [ 42%] library/email.iterators
reading sources... [ 42%] library/email.message

Exception occurred:
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/domains/python.py", line 568, in run
    (modname, self.env.doc2path(modules[modname])),
AttributeError: 'DirectiveAdapter' object has no attribute 'env'
The full traceback has been saved in /tmp/sphinx-err-16WiDR.log, if you want to report the issue to the developers.
￼
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [build] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
python, python@2: update resources https://github.com/Homebrew/homebrew-core/pull/27041
Change cookiecutter to depend on Python 3 https://github.com/Homebrew/homebrew-core/pull/27028
neovim: Add caveats for Python/Ruby support https://github.com/Homebrew/homebrew-core/pull/26869
Switch --with-python option for GDB formula to Python 3. https://github.com/Homebrew/homebrew-core/pull/24524
libgetdata: optional python and php bindings https://github.com/Homebrew/homebrew-core/pull/26223
libpeas: add support for running python plugins https://github.com/Homebrew/homebrew-core/pull/22430
gdb with --with-python@2 option builds against wrong python dir https://github.com/Homebrew/homebrew-core/issues/25230

Error: Your Xcode (9.2) is outdated.
Please update to Xcode 9.3 (or delete it).
Xcode can be updated from the App Store.
```

The error log's location and name /tmp/sphinx-err-16WiDR.log:
```
# Sphinx version: 1.7.3
# Python version: 2.7.10 (CPython)
# Docutils version: 0.14 
# Jinja2 version: 2.10
# Last messages:
#   reading sources... [ 40%] library/email.compat32-message
#   reading sources... [ 40%] library/email.contentmanager
#   reading sources... [ 41%] library/email.encoders
#   reading sources... [ 41%] library/email.errors
#   reading sources... [ 41%] library/email.examples
#   reading sources... [ 41%] library/email.generator
#   reading sources... [ 41%] library/email.header
#   reading sources... [ 42%] library/email.headerregistry
#   reading sources... [ 42%] library/email.iterators
#   reading sources... [ 42%] library/email.message
# Loaded extensions:
#   c_annotations (1.0) from /private/tmp/python-20180425-21380-mugcpu/Python-3.6.5/Doc/tools/extensions/c_annotations.py
#   sphinx.ext.coverage (1.7.3) from /usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/ext/coverage.py
#   sphinx.ext.doctest (1.7.3) from /usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/ext/doctest.py
#   alabaster (0.7.10) from /usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/alabaster/__init__.py
#   pyspecific (1.0) from /private/tmp/python-20180425-21380-mugcpu/Python-3.6.5/Doc/tools/extensions/pyspecific.py
Traceback (most recent call last):
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/cmdline.py", line 304, in main
    app.build(args.force_all, filenames)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/application.py", line 331, in build
    self.builder.build_update()
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 338, in build_update
    'out of date' % len(to_build))
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/builders/__init__.py", line 351, in build
    updated_docnames = set(self.env.update(self.config, self.srcdir, self.doctreedir))
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/environment/__init__.py", line 565, in update
    self._read_serial(docnames, self.app)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/environment/__init__.py", line 584, in _read_serial
    self.read_doc(docname, app)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/environment/__init__.py", line 659, in read_doc
    doctree = read_doc(self.app, self, self.doc2path(docname))
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/io.py", line 294, in read_doc
    pub.publish()
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/readers/__init__.py", line 72, in read
    self.parse()
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/parsers.py", line 85, in parse
    self.statemachine.run(inputstring, document, inliner=self.inliner)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 171, in run
    input_source=document['source'])
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 2753, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 395, in new_subsection
    node=section_node, match_titles=True)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 282, in nested_parse
    node=node, match_titles=match_titles)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 2326, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 2338, in explicit_construct
    return method(self, expmatch)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 2081, in directive
    directive_class, match, type_name, option_presets)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/vendor/lib/python2.7/site-packages/docutils/parsers/rst/states.py", line 2130, in run_directive
    result = directive_instance.run()
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/domains/__init__.py", line 237, in run
    return BaseDirective.run(self)
  File "/usr/local/Cellar/sphinx-doc/1.7.3/libexec/lib/python2.7/site-packages/sphinx/domains/python.py", line 568, in run
    (modname, self.env.doc2path(modules[modname])),
AttributeError: 'DirectiveAdapter' object has no attribute 'env'

```

#### Expected results
I expected Python3 to download.

### Reproducible project / your project
I don't have a project. I was just trying to download Python3.

### Environment info
- OS: Mac 10.13.4
- Python version: 2.7.10 (CPython)
- Sphinx version: 1.7.3

**Comment by jfbu** [comment]
This is presumably same as #4885, which is fixed at #4887 for soon to be released Sphinx 1.7.4.

In the meantime, it seems Homebrew has workaround see https://github.com/Homebrew/homebrew-core/issues/27026#issuecomment-384217523: do ``brew upgrade sphinx-doc``

**Comment by jfbu** [comment]
Closing as this was fixed already. Thanks for report!
